spring-cloud-azure-eventhub-stream-binder/src/test/java/com/microsoft/azure/eventhub/stream/binder/EventHubTestOperation.java

```diff
@@ -25,46 +24,35 @@
     private final Map<String, List<EventData>> eventHubsByName = new ConcurrentHashMap<>();
 
     @Override
-    public CompletableFuture<Void> sendAsync(String eventHubName, EventData data, PartitionSupplier partitionSupplier) {
+    public CompletableFuture<Void> sendAsync(String eventHubName, EventData message,
+            PartitionSupplier partitionSupplier) {
         CompletableFuture<Void> future = new CompletableFuture<>();
         eventHubsByName.putIfAbsent(eventHubName, new LinkedList<>());
-        eventHubsByName.get(eventHubName).add(data);
+        eventHubsByName.get(eventHubName).add(message);
         consumerMap.putIfAbsent(eventHubName, new ConcurrentHashMap<>());
-        consumerMap.get(eventHubName).values().forEach(c -> c.accept(Collections.singleton(data)));
+        consumerMap.get(eventHubName).values().forEach(c -> c.accept(Collections.singleton(message)));
         future.complete(null);
         return future;
     }
 
     @Override
-    public Subscriber<EventData> subscribe(String eventHubName, String consumerGroup) {
-        return new TestSubscriber(eventHubName, consumerGroup);
-    }
-
-    private class TestSubscriber implements Subscriber<EventData> {
-        private final String eventHubName;
-        private final String consumerGroup;
+    public boolean subscribe(String destination, Consumer<Iterable<EventData>> consumer, String consumerGroup) {
+        consumerMap.putIfAbsent(destination, new ConcurrentHashMap<>());
+        consumerMap.get(destination).put(consumerGroup, consumer);
+        eventHubsByName.putIfAbsent(destination, new LinkedList<>());
 
-        private TestSubscriber(String eventHubName, String consumerGroup) {
-            this.eventHubName = eventHubName;
-            this.consumerGroup = consumerGroup;
-        }
-
-        @Override
-        public void subscribe(Consumer<Iterable<EventData>> consumer) {
-            consumerMap.putIfAbsent(eventHubName, new ConcurrentHashMap<>());
-            consumerMap.get(eventHubName).put(consumerGroup, consumer);
-            eventHubsByName.putIfAbsent(eventHubName, new LinkedList<>());
-        }
+        return true;
+    }
 
-        @Override
-        public void unsubscribe() {
-            consumerMap.get(eventHubName).remove(consumerGroup);
-        }
+    @Override
+    public boolean unsubscribe(String destination, Consumer<Iterable<EventData>> consumer, String consumerGroup) {
+        consumerMap.get(destination).remove(consumerGroup);
+        return true;
+    }
 
-        @Override
-        public Checkpointer<EventData> getCheckpointer() {
-            return new EventHubTestCheckpointer();
-        }
+    @Override
+    public Checkpointer<EventData> getCheckpointer(String destination, String consumerGroup) {
```

Comment:
destination and consumerGroup are not used. why added?

Reply:
Same as above for testing purpose and simplicity.